src/test/java/com/google/sps/servlets/MatchesListServletTest.java

```diff
@@ -0,0 +1,174 @@
+    new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Mock
+  private HttpServletRequest mockRequest;
+ 
+  @Mock
+  private HttpServletResponse mockResponse;
+
+  private MatchesListServlet servletUnderTest;
+  private DatastoreService datastore;
+  private StringWriter responseWriter;
+  private PrintWriter writer;
+
+  @Before
+  public void setUp() throws IOException {
+    MockitoAnnotations.initMocks(this);
+
+    helper.setUp();
+
+    servletUnderTest = new MatchesListServlet();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  /**
+   * Tests the case in which a user's match list is empty.
+   *
+   * <p>Should return an empty list.
+   */
+  @Test
+  public void matchInfoWithEmptyList() throws IOException{
+    addMatchInfoEntityToDatastore(TEST_USER_ID, ImmutableList.of(),
+      ImmutableList.of(), ImmutableList.of(), ImmutableList.of());
+    
+    String actualOutput = execute(TEST_USER_ID);
+    
+    List<String> actualOutputAsList = Arrays.asList(new Gson().fromJson(actualOutput, String[].class));
+
+    assertThat(actualOutputAsList).isEqualTo(EMPTY_LIST);
+  }
+
+  /**
+   * Tests the case where a user has one match in their match list
+   *
+   * <p>Should return a list with just that one match's id
+   */
+  @Test
+  public void matchInfoWithSingleMatch() throws IOException{
+    addMatchInfoEntityToDatastore(TEST_USER_ID, ImmutableList.of(),
+      ImmutableList.of(), ImmutableList.of(), ImmutableList.of(TEST_CONNECTION_1_ID));
+    
+    String actualOutput = execute(TEST_USER_ID);
+    
+    List<String> actualOutputAsList = Arrays.asList(new Gson().fromJson(actualOutput, String[].class));
+
+    assertThat(actualOutputAsList).containsExactly(TEST_CONNECTION_1_ID);
+  }
+  
+  /**
+   * Tests the case where a user has three matches in their match list
+   *
+   * <p> Should a return a list with the three matches' ids
+   */
+  @Test
+  public void matchInfoWithMultipleMatches() throws IOException{
+    addMatchInfoEntityToDatastore(TEST_USER_ID, ImmutableList.of(), ImmutableList.of(), ImmutableList.of(),
+    ImmutableList.of(TEST_CONNECTION_1_ID, TEST_CONNECTION_2_ID, TEST_CONNECTION_3_ID));
+    
+    String actualOutput = execute(TEST_USER_ID);
+    
+    List<String> actualOutputAsList = Arrays.asList(new Gson().fromJson(actualOutput, String[].class));
+
+    assertThat(actualOutputAsList).containsExactly(TEST_CONNECTION_1_ID, TEST_CONNECTION_2_ID, TEST_CONNECTION_3_ID);
+  }
+
+  /**
+   * Method that calls on the MatchInformationServlet and returns the ID of the next
+   * potential match for the specified user
+   *
+   * @param userIDToFetch The ID of the user whose next potential match is being found
+   */
+  private String execute(String userIDToFetch) throws IOException{
+    when(mockRequest.getParameter(USER_ID_REQUEST_URL_PARAM)).thenReturn(userIDToFetch);
+    
```

Comment:
Remove.

Not as strictly a rule here, but I would read this method as "setup" -> "test" -> "assert" where you actually have "setup" -> "execute" -> "return". Grouping by blocks agin.